tools/run_tests/run_tests.py

```diff
@@ -488,19 +490,35 @@ def _get_pythons(self, args):
       builder_prefix_arguments = []
       venv_relative_python = ['bin/python']
       toolchain = ['unix']
+
       # Bit-ness is handled by the test machine's environment
       python_pattern_function = lambda major, minor, bits: 'python{major}.{minor}'.format(major=major, minor=minor)
+
+      pypy_pattern_function = lambda major: 'pypy{major}'.format(major=major) if major == 3 else 'pypy'
```

Comment:
Just define `_pypy_pattern(major):` as a module-scope function (even though it will have a one-line body and be used in only one place).